fieldsolver/ldz_magnetic_field.cpp

```diff
@@ -86,12 +86,20 @@ void propagateMagneticField(
             EGrid2 = EGrid.get(i,j,k+1);
             perBDt2Grid0->at(fsgrids::bfield::PERBX) = perBGrid0->at(fsgrids::bfield::PERBX) + 0.5*dt*(1.0/dz*(EGrid2->at(fsgrids::efield::EY) - EGrid0->at(fsgrids::efield::EY)) + 1.0/dy*(EGrid0->at(fsgrids::efield::EZ) - EGrid1->at(fsgrids::efield::EZ)));
             break;
-            
+
+         case RK_ORDER2_STEP1SS:
+            perBDt2Grid0 = perBDt2Grid.get(i,j,k);
+            EGrid0 = EGrid.get(i,j,k);
+            EGrid1 = EGrid.get(i,j+1,k);
+            EGrid2 = EGrid.get(i,j,k+1);
+            perBDt2Grid0->at(fsgrids::bfield::PERBX) += dt/dz*(EGrid2->at(fsgrids::efield::EY) - EGrid0->at(fsgrids::efield::EY)) + dt/dy*(EGrid0->at(fsgrids::efield::EZ) - EGrid1->at(fsgrids::efield::EZ));
```

Comment:
This (and the next two similar lines) should be `=` and not `+=` I think. Compare with line 87.

Reply:
No, I think this is as intended. Line 87, for the first of a group of substeps, evaluates perB_dt2 as a half-step ahead from perB.
This line evaluates the n>1 substep perB_dt2 as a full-timestep advance from the perB_dt2 of the previous substep.

I'll check out what's happening with boundary conditions though as that's where the blowup was located.